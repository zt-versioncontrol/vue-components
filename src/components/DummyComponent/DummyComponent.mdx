import { Meta, ArgTypes } from '@storybook/blocks'
import DummyComponent from '@/components/DummyComponent/DummyComponent.vue'
import * as Stories from './DummyComponent.stories.ts'
import {
  ComponentSummary,
  CustomTypes,
  ComponentDependency,
  Examples,
  ExampleGroup
} from '/.storybook/DocBlocks'

<Meta title="DummyComponent Document" of={Stories} component={DummyComponent} />

<ComponentSummary of={Stories} />

<ArgTypes of={Stories} />

<CustomTypes of={Stories} />

<ComponentDependency of={Stories} />


<Examples>
  # TODO: define example groups
</Examples>
