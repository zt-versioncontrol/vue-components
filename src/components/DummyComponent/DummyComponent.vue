<script setup lang="ts">

// TODO: declare props
// TODO: add documentation comment for each prop
// TODO: if a prop is an object with specific attributes or a string with limited set of values then define and export a type
export interface Props {
}

// TODO: define props
const props = withDefaults(defineProps<Props>(), {
})

// TODO: define emits
// TODO: add documentation comment for each emit
// TODO: if an event emits an object with specific attribures or a string with a limited set of values then define and export a type
const emit = defineEmits<{
}>()

// TODO: define slots
// TODO: add documentation comment for each slot
// TODO: if a slot passes a prop that is an object with specific set of attribures or a string with a liited set of values then define and eport a type
defineSlots<{
}>()

// TODO: define exposed attributes and methods
// TODO: add documentation comment for each item
defineExpose({
})


// TODO: add any necessary code here

</script>

<!-- TODO: define component template -->
<template>
  <h1>DummyComponent - waiting for implementation</h1>
</template>

<!-- TODO: define styles -->
<style scoped>
</style>

<!-- TODO: define internationalization strings -->
<i18n lang="yaml">
</i18n>
