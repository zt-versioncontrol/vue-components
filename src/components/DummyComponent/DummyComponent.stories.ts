import type { Meta, StoryObj } from '@storybook/vue3'
import DummyComponent from '@/components/DummyComponent/DummyComponent.vue'
import { type Documentation } from '@/../.storybook/DocUtilities'


const meta: Meta<typeof DummyComponent> = {
  // TODO: define components' storybook path using title attribute
  title: '',
  component: DummyComponent,
  excludeStories: ['documentation']
}

export default meta

type Story = StoryObj<typeof DummyComponent>

//TODO: fill in the documentation
export const documentation: Documentation = {
}

// TODO: add all necessary stories
export const Story1: Story = {
  args: {
  }
}
