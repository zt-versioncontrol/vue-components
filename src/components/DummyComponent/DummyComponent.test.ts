import { mount } from '@vue/test-utils'
import { describe, it, expect } from 'vitest'
import DummyComponent from '@/components/DummyComponent/DummyComponent.vue'


describe('Default component', () => {
  it('should be rendered', () => {
    const wrapper = mount(DummyComponent, {
      global: {
        // TODO: add required plugins
        plugins: []
      },
      attachTo: document.body
    })

    expect(wrapper.isVisible()).toBe(true)
  })

  it('should match snapshot', () => {
    const wrapper = mount(DummyComponent, {
      global: {
        // TODO: add required plugins
        plugins: []
      }
    })

    expect(wrapper.element).toBeTruthy()
    expect(wrapper.element).toMatchSnapshot()
  })
})

// TODO: add test groups as necessary
describe('test group1', () => {
  // TODO:
  it.todo('should pass this test', () => {
  })
  // TODO:
  it.todo('should pass that test', () => {
  })
})
