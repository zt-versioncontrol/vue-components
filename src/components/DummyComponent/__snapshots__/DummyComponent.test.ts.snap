// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Default component > should match snapshot 1`] = `
<h1
  data-v-2eea0293=""
>
  DummyComponent - waiting for implementation
</h1>
`;
